generators/example/src/main/scala/ConfigMixins.scala

```diff
@@ -33,23 +34,26 @@ import ConfigValName._
 // Common Parameter Mixins
 // -----------------------
 
-/**
- * Class to specify where the BootRom file is (from `rebar` top)
- */
 class WithBootROM extends Config((site, here, up) => {
   case BootROMParams => BootROMParams(
     contentFileName = s"./bootrom/bootrom.rv${site(XLen)}.img")
 })
 
-// DOC include start: WithGPIO
-/**
- * Class to add in GPIO
- */
+// DOC include start: gpio mixin
 class WithGPIO extends Config((site, here, up) => {
   case PeripheryGPIOKey => Seq(
     GPIOParams(address = 0x10012000, width = 4, includeIOF = false))
+  case BuildTop => (clock: Clock, reset: Bool, p: Parameters, success: Bool) => {
+    val top = up(BuildTop, site)(clock, reset, p, success)
+    for (gpio <- top.gpio) {
+      for (pin <- gpio.pins) {
+        pin.i.ival := false.B
```

Comment:
Is this tying all of the GPIO inputs to 0 inside the DUT?

Reply:
This code was copied from the older `WithGPIOTop` mixin. I'm not sure what the right thing to do here is, but I would be happy to implement it

Reply:
This is called from outside the DUT (in the TestHarness), so it should be tying them to 0 outside the DUT.

Reply:
I think we want a separate PR that adds IO cell blackboxes to have "Correct" functionality. That's been a work in progress for a while, but we can reprioritize it.

Reply:
While I agree with John on this, I don't think we need to block on this? Also, it might be good to add a comment that this is connecting from outside the DUT.

Reply:
> separate PR 

means it's not blocking